packages/engine/src/datastore/batch/agent.rs

```diff
@@ -68,21 +51,23 @@ impl AgentBatch {
     }
 
     pub fn duplicate_from(
-        batch: &Self,
+        agent_group: &Self,
```

Comment:
I'd prefer if we had this as `agent_batch` until we do the TODO above to rename the actual object. I know it'll look bad with `agent_batch.batch...` but at least it's consistent.

Reply:
Having thought more, I think it might also be misleading because a Group is actually a combination of an AgentBatch and MessageBatch, and everywhere we use `agent_group` in this PR doesn't really make sense..

Reply:
Well, a Group could be a combination of an AgentGroup and a MessageGroup

Reply:
A combination of `AgentBatch` and `MessageBatch` is a `State`. As we want to remove the term `Batch` from `AgentBatch` and `MessageBatch`, a combination of `AgentGroup` and `MessageGroup` is a `StateGroup`. An `AgentPool` is then a collection of `AgentGroup` (agent group**s**). 

Anyway, I agree on keeping `agent_batch.batch` as long as we have an `AgentBatch` struct to avoid confusion.

Reply:
I don't think that's the right definition at all

State is made up of lots of things, within State you have a Pool of Groups of Agents, within those Groups you have Messages data and AgentState data. Both of those live inside their own batches and memory segments

Reply:
An Agent is made up of a collection of data, some of which lives within the Messages batches, and some within the AgentState batches. But a Group is a collection of Agents, it's not a collection of messages or a collection of AgentState.

Reply:
Agreed, I just didn't want the comments persisted on this PR to read as if that was the way of things and has been decided. The point is we still need to discuss it and figure it out :D